massa-module-cache/gas-table.md

```diff
@@ -0,0 +1,61 @@
+# Overview
+
+This document keeps track of where and how gas costs are handled for SC executions.
+
+It contains:
+* the gas table
+* the list of functions that effectively pay the costs
+
+# Gas table
+
+| TYPE                     | PAYMENT                             |
+| ------------------------ | ----------------------------------- |
+| **CallSC (OP)**          |                                     |
+| compilation              | NOT PAID                            |
+| VM instantiation         | PAID IN MAX_GAS (UNIFIED CONSTANT)  |
+| module instantiation     | PAID IN MAX_GAS                     |
+| execution                | PAID IN MAX_GAS                     |
+| **ExecuteSC (OP)**       |                                     |
+| compilation              | PAID IN MAX_GAS                     |
+| VM instantiation         | PAID IN MAX_GAS (UNIFIED CONSTANT)  |
+| module instantiation     | PAID IN MAX_GAS                     |
+| execution                | PAID IN MAX_GAS                     |
+| **Call (ABI)**           |                                     |
+| compilation              | NOT PAID                            |
+| VM instantiation         | PAID IN CALL GAS (UNIFIED CONSTANT) |
+| module instantiation     | PAID IN CALL GAS                    |
+| execution                | PAID IN CALL GAS                    |
+| base gas cost of the ABI | PAID IN ABI GAS                     |
+| **LocalExecution (ABI)** |                                     |
+| compilation              | PAID IN CALL GAS                    |
```

Comment:
Wouldn't it be more consistent if this part is paid in the ABI gas ? note that it is equivalent in practice

Reply:
Yes it would but that means we need to explicitly pay what need to be payed for anything that uses tmp modules. Whether its in the ABI costs or execution.rs we would have to never forget about it, with the current system anything that wants to use a tmp module automatically pays and checks compil & instance cost, since there are no cases where we want to do one but not the other. I agree that changing the ABI costs would make it more consistent when looking at the system from outside but inside it actually makes it more complex and prone to error. I have no strong opinion on this if you prefer one or the other let me know I'll change it just wanted you to be aware of that.

Reply:
Okay I understand your point. Then let's keep it as is

Reply:
After further thought on your previous comments, if we move the compilation base cost of Execute SC towards the constant operation cost, this should be changed as well to move towards the ABI cost

Reply:
So we keep it as is yes?

Reply:
Or do we move the compilation cost to OP and ABI costs respectively?

Reply:
I would vote for moving it to OP/ABI to prioritize user-facing consistency over code factorization

Reply:
Ok